util/weeklog.js

```diff
@@ -46,8 +49,11 @@ function parseLog (buffer, categoryList) {
       entry.timestamp += buffer[curr + 14 + j] * Math.pow(256, 2 - j);
     }
 
-    // this pattern marks the removal of an entry
-    if (entry.time === 0 && entry.portals === 0) {
+    // a time of zero logs the removal of the previous entry
+    if (entry.time === 0) {
+
+      // the portal count byte denotes whether to purge the run from the output entirely
+      if (!forcePurge && entry.portals === 0) continue;
```

Comment:
I find this to be pretty hard to read, but I also don't know this code too much, so as long as there's a good reason why you're doing it this way I guess I won't complain